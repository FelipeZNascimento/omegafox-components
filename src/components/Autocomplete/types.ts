export type TDropdownItem = {
  id: number;
  name: string;
  abbreviationEn: string;
};

export interface IAutocompleteProps {
  dropdownList: TDropdownItem[];
  onSelect: (item: TDropdownItem) => void;
}
