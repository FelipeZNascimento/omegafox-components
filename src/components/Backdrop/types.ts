export interface IBackdropProps {
  align?: 'left' | 'center' | 'right';
  children: JSX.Element;
  isOpen: boolean;
}
