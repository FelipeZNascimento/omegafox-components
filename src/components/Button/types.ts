export interface IButtonProps extends React.HTMLAttributes<HTMLButtonElement> {
  children?: React.ReactNode | null;
  icon?: React.ReactNode | null;
  isDisabled?: boolean;
  isDarkMode?: boolean;
  isSelected?: boolean;
  leftIcon?: boolean;
  size?: 'small' | 'medium' | 'large';
  shape?: 'rounded' | 'squared';
  shadowed?: boolean;
  tooltip?: string;
  variant?: 'primary' | 'danger' | 'neutral' | 'confirm' | 'warning';
  onClick: () => void;
}
