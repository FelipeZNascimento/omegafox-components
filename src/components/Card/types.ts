export interface ICardProps {
  isForceMobile?: boolean;
  isSelected: boolean;
  title: string;
  subtitle?: string | null;
  onClick?: null | (() => void);
  renderingStatusFunction?: null | (() => JSX.Element | null);
}
