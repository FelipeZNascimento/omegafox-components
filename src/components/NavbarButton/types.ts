export type TNavbarButton = {
  id: number;
  isDisabled?: boolean;
  text: string;
  url: string;
  withCredentials?: boolean;
  renderingFunction?: () => JSX.Element;
};

export type TNavbarButtonProps = {
  button: TNavbarButton;
  isSelected: boolean;
  isShadowed?: boolean;
  onClick: (button: TNavbarButton) => void;
};
