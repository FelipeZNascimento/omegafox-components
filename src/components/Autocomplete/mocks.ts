import { TDropdownItem } from './types';

export const dropdownList: TDropdownItem[] = [
  {
    id: 0,
    name: 'Lionel Messi',
    abbreviationEn: 'ARG'
  },
  {
    id: 1,
    name: 'Cristiano Ronaldo',
    abbreviationEn: 'POR'
  },
  {
    id: 2,
    name: 'Mbappé',
    abbreviationEn: 'FRA'
  },
  {
    id: 3,
    name: 'Caio Jr.',
    abbreviationEn: 'BRA'
  },
  {
    id: 4,
    name: 'Darwin Nuñez',
    abbreviationEn: 'URU'
  }
];
