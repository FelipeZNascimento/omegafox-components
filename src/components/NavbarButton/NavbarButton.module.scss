@import "../../styling/alignments.scss";
@import "../../styling/colors.scss";
@import "../../styling/font-sizes.scss";
@import "../../styling/margins.scss";
@import "../../styling/paddings.scss";

.navButton {
  @include font-sized(navbar-button);
  @include padded(all, s);
  @include marged(horizontal, s3);

  border-radius: 16px;
  height: 100%;

  &Shadowed {
    @include shadowed($color-grey3, 0.2);
  }

  &Enabled {
    @include font-colored($color-grey4);

    transition: background-color $transition-time, color $transition-time;
    cursor: pointer;

    &Selected {
      @include font-colored($color-qatar-red);
    }

    &:hover {
      @include bg-colored($color-grey2);

      transition: background-color $transition-time, color $transition-time;
    }
  }

  &Disabled {
    @include font-colored($color-grey3);

    cursor: not-allowed;
    pointer-events: all !important;
  }
}
