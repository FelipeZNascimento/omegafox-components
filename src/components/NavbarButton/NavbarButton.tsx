import React from 'react';

import classNames from 'classnames';
import { TNavbarButtonProps } from './types';
import styles from './NavbarButton.module.scss';

export const NavbarButton = ({
  button,
  isFlexed = false,
  isSelected,
  isShadowed = false,
  theme,
  onClick
}: TNavbarButtonProps) => {
  const titleCase = (string: string) => {
    return string[0].toUpperCase() + string.slice(1).toLowerCase();
  };

  if (button.renderingFunction) {
    return (
      <button
        style={{ backgroundColor: 'transparent' }}
        key={button.id}
        onClick={() => onClick(button)}
      >
        {button.renderingFunction()}
      </button>
    );
  }
  const classTheme = `navButton${titleCase(theme)}`;
  const buttonClass = classNames(styles.navButton, {
    [styles[`${classTheme}Enabled`]]: !button.isDisabled,
    [styles.navButtonDisabled]: button.isDisabled,
    [styles[`${classTheme}EnabledSelected`]]: isSelected,
    [styles[`${classTheme}Shadowed`]]: isShadowed,
    [styles.navButtonFlexed]: isFlexed
  });

  return (
    <button
      key={button.id}
      className={buttonClass}
      onClick={() => onClick(button)}
    >
      {button.text}
    </button>
  );
};
