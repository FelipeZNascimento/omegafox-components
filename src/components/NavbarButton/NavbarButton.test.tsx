import React from 'react';
import { render } from '@testing-library/react';
import { NavbarButton } from './NavbarButton';
import { TNavbarButtonProps } from './types';
import { navbarButtonsLeft } from 'components/mocks';

describe('<NavbarButton />', () => {
  const renderComponent = ({ button, isSelected }: TNavbarButtonProps) =>
    render(
      <NavbarButton
        button={button}
        isSelected={isSelected}
        theme="copa"
        onClick={jest.fn}
      />
    );

  it('should render', () => {
    const { container } = renderComponent({
      button: navbarButtonsLeft[0],
      isSelected: true,
      theme: 'copa',
      onClick: () => console.log('Clicked')
    });

    expect(container).toMatchSnapshot();
  });
});
