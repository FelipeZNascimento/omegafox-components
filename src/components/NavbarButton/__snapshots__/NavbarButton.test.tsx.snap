// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavbarButton /> should render 1`] = `
<div>
  <button
    class="navButton navButtonCopaEnabled navButtonCopaEnabledSelected"
  >
    In√≠cio
  </button>
</div>
`;
